mautic/mautic
# TPROD-327 Twig refactoring of the SMS bundle

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://mautic.atlassian.net/browse/TPROD-327



#### Description:

Replaces PHP templates with Twig templates for the Sms Bundle which brings us one more step closer to Symfony 5 upgrade.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Enable the Twilio plugin with any bogus credentials. 
3. You should be able to see the Channels > Text Messages menu
4. Click through the UI to see if it works as before.

To test the contact detail timeline event I created this row:
```sql
INSERT INTO `mautic_sms_message_stats` (`id`, `sms_id`, `lead_id`, `list_id`, `ip_id`, `date_sent`, `is_failed`, `tracking_hash`, `source`, `source_id`, `tokens`, `details`)
VALUES
	(1, 1, 1, 1, NULL, '2023-01-01 00:00:00', 1, 'sfasfasdfas', NULL, NULL, NULL, '{}');
```
A contact with ID 1 and SMS with ID 1 must exist. If not either create them via UI or change the IDs in the query. It renders like this on the contact detail page:
![Screen Shot 2023-01-12 at 16 36 50](https://user-images.githubusercontent.com/1235442/212111256-642fccb0-a3b6-4f25-b0dd-258979098d35.png)

A known issue is that the title for the global config for the SMS disappears. That will be fixed when the Config Bundle is converted to Twig as well. Another issue I found was that the select box shows untranslated string. But that is on the 5.x branch the same so I'm leaving is as is as it was not introduced by this PR.
